guides/troubleshooting/docker-problems.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Docker problems
+description: Learn how to resolve issues related to Docker inside Coder workspaces.
+---
+
+When using Coder, you may encounter the following error:
+
+```console
+docker: Error response from daemon: OCI runtime create failed:
+container_linux.go:370: starting container process caused:
+process_linux.go:459: container init caused: join session keyring:
+create session key: disk quota exceeded: unknown.
+```
+
+## Why this happens
+
+The kernel allocates a system key for each container created. When lots of
+developers are sharing the same instance, you may run into limits on the number
+and size of keys each user can have.
+
+## Resolution
+
+To fix this error, you can increase `maxkeys` and `maxbytes`. These are global
+settings that apply to *all* users sharing the same system. You can modify this
+by adding the following to the `sysctl` configuration file:
+
+```console
+sudo sysctl -w kernel.keys.maxkeys=20000
+sudo sysctl -w kernel.keys.maxbytes=400000
+```
+
+Alternatively, you can use a DaemonSet with `kubectl apply` to make changes to
+`sysctl`:
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: increase-limits
+  namespace: kube-system
+  labels:
+    app: increase-limits
+    k8s-app: increase-limits
+spec:
+  selector:
+    matchLabels:
+      k8s-app: increase-limits
+  template:
+    metadata:
+      labels:
+        name: increase-limits
+        k8s-app: increase-limits
+      annotations:
+        seccomp.security.alpha.kubernetes.io/defaultProfileName: runtime/default
+        apparmor.security.beta.kubernetes.io/defaultProfileName: runtime/default
+    spec:
+      nodeSelector:
+        kubernetes.io/os: linux
+      initContainers:
+        - name: sysctl
+          image: alpine:3
+          command:
+            - sysctl
+            - -w
+            - kernel.keys.maxkeys=100000003
```

Comment:
Can we use the same numbers here that we do above for the sysctl method? `sysctl -w kernel.keys.maxkeys=20000` - just makes things internally consistent in the doc

```suggestion
            - kernel.keys.maxkeys=20000
```